### PR TITLE
Bugfix/multiple redirections on my account refresh

### LINKED ISSUE #2713: Infinite loop on multi-store site after Account page reload
## Current behavior

Reloading any my-account url on a multi-store site causes an infinite redirect loop and constant requests to the server

## Expected behavior

A single reload

## Steps to reproduce the issue

1. In a browser tab, go to https://demo.storefrontcloud.io/de/
2. Login (roni_cost@example.com, roni_cost3@example.com)
3. Navigate to any Account page, eg My Orders
4. Reload the page using the browser Reload button
5. Observe the infinite reload

Make sure to close the browser tab to prevent server overload!

## Can you handle fixing this bug by yourself?

- [ ] YES
- [x] NO

Well, maybe. I'm trying but haven't found a proper fix yet and don't have any more time today.

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [x] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

## Environment details

- Browser: Chrome 73
- OS: Mac OS
- Node: 8.11.4
- Code Version: 1.8.4, 1.9

## Additional information

The problem is in vue-storefront/core/modules/user/router/beforeEach.ts
```
next('/')
setTimeout(()=> {
  router.push(to.path)
}, 0)
```

Redirecting to '/' on a multi-store site always causes problems. I'm not sure exactly what's happening here, but I'm pretty sure this is where the problem is.

In this case, the current store is not set at this stage, so using `localizedRoute('/', currentStoreView().storeCode)` is not useful.

### PR DESCRIPTION
**Pull request by patzick**
### Related issues

closes #2713 

### Short description and why it's useful

Fixes problem with infinite redirections and redirections on SSR on account site. What has been done:
- changed `user/startSession` to async method and awaits its invocations
- make sure in `router/beforeEach` that local session data are readed before redirecting to home page
- do not redirect to home on SSR render on auth site, using NoSSR for session-related data and no more blinking home->account (and no hydration issue with that change)
- startSession only once with reading localStorage data, there were many requests with multiple invocations and slowed down initial render
- user data in localstorage are kept on single key, removed prefix causing problem with reading local session after changing language

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [x] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [ ] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [ ] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/user/store/getters.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ const getters: GetterTree<UserState, RootState> = {
   isLoggedIn (state) {
     return state.current !== null
   },
```
Comment (pkarw) [inline core/modules/user/store/getters.ts]: It should be named like “local_session_started” or “local_session_loaded”

File: core/modules/user/store/mutation-types.ts [hunk 2]
```diff
@@ -7,4 +7,5 @@ export const USER_START_SESSION = SN_USER + '/START_SESSION'
 export const USER_END_SESSION = SN_USER + '/END_SESSION'
 export const USER_UPDATE_PREFERENCES = SN_USER + '/UPDATE_PREFERENCES'
 export const USER_GROUP_TOKEN_CHANGED = SN_USER + '/GROUP_TOKEN_CHANGED'
-export const USER_GROUP_CHANGED = SN_USER + 'GROUP_ID_CHANGED'
\ No newline at end of file
+export const USER_GROUP_CHANGED = SN_USER + 'GROUP_ID_CHANGED'
```
Comment (pkarw) [inline core/modules/user/store/mutation-types.ts]: / - missing char in the mutation name

**Comment by pkarw** [comment]
Cool! Merged it

**Comment by michhy** [comment]
Hello,
I have tested it is working (solving the refresh issue) on default theme,
but not the vuetique theme.

Any idea of the related file that causing the difference?
thanks

**Pull request merged**